src/entity/CommonGame.java

```diff
@@ -61,41 +59,45 @@ public int getRoundsPlayed() {
     }
 
     @Override
-    public int getScore() {return score;}
-    @Override
-    public Round getCurrentRound() {return allRounds.get(allRounds.size() - 1);}
+    public int getScore() { return score; }
+
     @Override
-    public List<Round> getRounds() {
-        return allRounds;
+    public Round getCurrentRound() {
```

Comment:
I am a little confused on how this function will work with multiple rounds. In this function we are getting the last element in the round list which we are calling the current round. This would mean we create a round and a append that round after we complete a round. However, when we create the game object I believe we are going to add all the rounds at once which will allow us to load/continue a game as well as not have a wait time between rounds to create the next round which would need to have an api call. Correct me if I am mistaken though as I know we are still working out some aspects of this :)

Reply:
My interpretation of this implementation is that we would only create rounds as needed instead of all at once. So in this implementation `setCurrentRound` would be responsible for adding a round to the end of the `allRounds` list which would make `getCurrentRound` work properly. However, this is just one implementation. We also can make a different implementation that follows the approach you describe here - there are pros and cons to both :)

Reply:
Ah ok, yes that makes sense. I thought the implementation was to create a game and add all rounds at once. There are def pros and cons to both. Lets discuss this in more detail soon, but for now this pr is looking great good job.